AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Gamemode_Timing_Test_Template.py

```diff
@@ -0,0 +1,133 @@
+    the level loads, can enter/exit gameplay, and that Editor.exe remains stable throughout this process.
+
+    Test setup:
+    - Launch Editor.exe
+
+    Expected Behavior:
+    Test verifies that level loads, enters/exits game mode, and editor remains stable.
+
+    Test Steps:
+    1) Load level, confirm that correct level is loaded, and report results
+    2) Start timer for enter game mode.
+    3) Validate that editor can enter gameplay successfully
+    4) Checks if the timeout has been reached (default 10 sec), if not report time editor took to enter game mode.
+    5) Start timer for exit game mode
+    6) Validate that editor can exit gameplay successfully
+    7) Checks if the timeout has been reached (default 10 sec), if not report time editor took to exit game mode.
+    8) Open an empty level to verify editor remains stable
+    9) Look for errors or asserts.
+
+    :return: None
+    """
+
+    import azlmbr.legacy.general as general
+    import time
+
+    from editor_python_test_tools.utils import Report, Tracer, TestHelper
+
+    # Timer class to track enter/exit game time
+    class Timer:
+        unit_divisor = 60
+        hour_divisor = unit_divisor * unit_divisor
+
+        def start(self):
+            self._start_time = time.perf_counter()
+
+        def stop(self):
+            self._stop_time = time.perf_counter() - self._start_time
+            return self._stop_time
+
+        def log_time(self, message):
+            elapsed_time = time.perf_counter() - self._start_time
+            hours = int(elapsed_time / Timer.hour_divisor)
+            minutes = int(elapsed_time % Timer.hour_divisor / Timer.unit_divisor)
+            seconds = elapsed_time % Timer.unit_divisor
+
+            return(f'{message}: {hours:0>2d}:{minutes:0>2d}:{seconds:0>5.2f}\n')
+
+        def just_seconds(self):
+            elapsed_time = time.perf_counter() - self._start_time
+            seconds = elapsed_time % Timer.unit_divisor
+
+            return seconds
+
+    with Tracer() as error_tracer:
+
+        # 1. Load target level, confirm that correct level is loaded, and report results.
+        TestHelper.init_idle()
+        TestHelper.open_level("Graphics", Target_Level)
+        Report.result(Tests.target_level_load, Target_Level
+                      == general.get_current_level_name())
+
+        # 2. Start timer for enter game mode.
+        enter_gamemode_length = Timer()
+        enter_gamemode_length.start()
+
+        # 3. Validate that editor can enter gameplay successfully.
+        TestHelper.enter_game_mode(Tests.target_level_enter_game_mode)
+
+        # 4. Checks if the timeout has been reached (default 10 sec), if not report time editor took to enter game mode.
+        Report.critical_result(Tests.timeout_status, enter_gamemode_length.stop() < 10.0)
+
+        Report.success((enter_gamemode_length.log_time
```

Comment:
this calls a different time than .stop() that is more time has elapsed

Reply:
I tested it and it does not, it is the exact same

Reply:
also you just mentioned above that I should remove stop... and just use log time...?

Reply:
every time you call time.perf_counter() it's getting the time now. each one of those methods is calling it again and subtracting the start time so depending on how long various lines take could be different times.
because each method is not connected to a definitive stop, it seemed like you didn't need both, but on further reading of the script I see that each system is used independently